src/CoreMidi/MidiServices.cs

```diff
@@ -1189,11 +1193,13 @@ public MidiEndpoint GetSource (nint idx)
 			}
 		}
 
-		public NSData ConnectionUniqueIDData {
+		public NSData? ConnectionUniqueIDData {
 			get {
 				return GetData (kMIDIPropertyConnectionUniqueID);
 			}
 			set {
+				if (value is null)
+					ObjCRuntime.ThrowHelper.ThrowArgumentNullException (nameof (value));
```

Comment:
While I realize that I added a good amount of these and that SetData () throws an exception if the value arg is null, I thought it would be better to explicitly add the exceptions here and not use a `!`, but I am open to opinions!

Reply:
Explicit over implicit, I like it.